samples/java/jre/console/src/com/microsoft/cognitiveservices/speech/samples/console/SpeechRecognitionSamples.java

```diff
@@ -356,9 +356,18 @@ else if (e.getResult().getReason() == ResultReason.NoMatch) {
             // Push audio read from the file into the PushStream.
             // The audio can be pushed into the stream before, after, or during recognition
             // and recognition will continue as data becomes available.
-            while(  inputStream.read(readBuffer) != -1)
+            int bytesRead;
+            while ((bytesRead = inputStream.read(readBuffer)) != -1)
             {
-                pushStream.write(readBuffer);
+                if (bytesRead == readBuffer.length)
+                {
+                    pushStream.write(readBuffer);
+                }
+                else
+                {
+                    // Last buffer read from the WAV file is likely to have less bytes
```

Comment:
>is likely [](start = 58, length = 9)

This must be guaranteed to be less than readBuffer.length?

Reply:
Yes, per the link below FileInputStream.read(byte[] b) "Reads up to b.length bytes of data from this input stream into an array of bytes. This method blocks until some input is available"
https://docs.oracle.com/javase/7/docs/api/java/io/FileInputStream.html